### DISCUSSION TITLE
Missing standard texture components

### DISCUSSION THREAD

**Discussion by NovakEvana** [mtex-toolbox/mtex#1653] [Ask Anything] [answered]
Hello everyone,

I think I'll separate this problem into a new thread from the [original one](https://github.com/mtex-toolbox/mtex/discussions/1640). Somehow this becomes a bigger issue than I thought when I was playing around with texture components. The texture components listed in my code below were supposed to be missed at some places in both phi2 = 45 (yet to check another value) when they were compared to another "standard" reference from a publication. Also, many thanks to @kilir, I don't use `preserveOrder` anymore in my code. 

Thank you to everyone who is running and contributing to this community. I'm really thankful for that!

### **My code:**
```
cs = crystalSymmetry('m-3m');
ss = specimenSymmetry('triclinic');
close all;
phi_2 = [45]
figure;
hold on;   
plotSection(orientation.cube(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'purple','Marker','s');
plotSection(orientation.cubeND45(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'blue','Marker','s');
plotSection(orientation.copper(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'purple','Marker','<');
plotSection(orientation.twincopper(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'blue','Marker','<');
plotSection(orientation.gossND(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'blue','Marker','o');
plotSection(orientation.gossbrass(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'green','Marker','o');
plotSection(orientation.brass(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'purple','Marker','>');
plotSection(orientation.D(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'green','Marker','<');
plotSection(orientation.Y(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'purple','Marker','v');
plotSection(orientation.Z(cs,ss), 'phi2',[phi_2]*degree,'add2all','MarkerColor', 'purple','Marker','^');
hold off;
```

### **My results:**
![image](https://user-images.githubusercontent.com/117882416/201597314-960a9f37-8545-4f1d-a045-f30465ada2aa.png)

### **"Standard" reference from paper:**
![image](https://user-images.githubusercontent.com/117882416/201595817-c532eaa3-0d22-491b-a1ac-8e6c2014cce5.png)

**Comment by NovakEvana** [comment]
Hi Ralf and Rudiger, could I ask for your idea on this?

**Reply by trudator** [reply]
See somebody is having the same finding. I think Evana raised a fair question. Playing around with the standard orientation, such as cube or cube_ND, and found it was missing at 45 degrees. Not sure if something is wrong with the crystal or sample symmetry. 

![image](https://user-images.githubusercontent.com/117386810/203435386-22c16959-527f-49f2-9a23-9ac16b22615e.png)

**Comment by ralfHielscher** [comment, marked as answer]
Hi Evana,

thank you for bringing up this topic. I had a look into it and can confirm that there was a bug in MTEX. I hope I was able to fix it in this commit 87a81de5622f57850472167d09479d8d86496ed8

As you can see MTEX is now able to reproduce the picture in your publication:
![standardOri](https://user-images.githubusercontent.com/8619241/207390719-26e68cf9-2612-4e83-b263-a6639b907547.png)

However, the are a few things that still do not fit:
1. Your reference paper seems to have orthorhombic specimen symmetry
2. The definition of the alpha fibre in your reference paper is clearly wrong - its the same as the gamma fibre.
3. The orange fibre in the plot is <110> || Z, however this does not coincide with the definition of the alpha fibre within MTEX.
4. The tau fibre as define in your reference paper does not coincide with the definition of the tau fibre within MTEX. 

Is there any reliable source on how these fibres are defined?

Below the code I used.

All the best, Ralf.

```matlab
opt = {'phi2',phi_2,'add2all','MarkerEdgeColor','black','maxphi1',2*pi,'MarkerSize',20};

ori = orientation.cube(cs,ss);
plotSection(ori, 'MarkerColor', 'darkred','Marker','s', opt{:});
hold on; 

ori = orientation.cubeND45(cs,ss)
plotSection(ori,'MarkerColor', 'steelblue','Marker','d', opt{:});

gamma = fibre.gamma(cs)
plot(gamma,'add2all','lineColor','green','LineWidth',3,'LineStyle','--');

alpha = fibre(Miller(1,1,0,cs),zvector)
plot(alpha,'add2all','lineColor','orange','LineWidth',3,'LineStyle',':')

%tau = fibre.tau(cs)
tau = fibre(Miller(1,1,0,cs),yvector)
plot(tau,'add2all','lineColor','red','LineWidth',3,'LineStyle',':');

ori = orientation.byMiller([1 1 1],[1 1 2],cs);
plotSection(ori,'MarkerColor', 'gold','Marker','h',opt{:});

ori1 = orientation.byMiller([1 1 0],[1 1 1],cs);
plotSection(ori1,'MarkerColor', 'lightSkyBlue','Marker','h',opt{:});

ori2 = orientation.byMiller([1 1 0],[1 1 2],cs);
plotSection(ori2,'MarkerColor', 'r','Marker','d',opt{:});

ori = orientation.byMiller([1 1 0],[0 0 1],cs);
plotSection(ori,'MarkerColor', 'g','Marker','o',opt{:});

% 
ori = orientation.byMiller([5 5 2],[1 1 5],cs);
plotSection(ori,'MarkerColor', 'b','Marker','+',opt{:});

% ^
ori = orientation.byMiller([1 1 2],[1 1 1],cs);
plotSection(ori,'MarkerColor', 'none','Marker','^',opt{:});

% v
ori = orientation.byMiller([4 4 11],[11 11 8],cs);
plotSection(ori,'MarkerColor', 'none','Marker','v',opt{:});

hold off
```

**Reply by NovakEvana** [reply]
Hi Ralf,

Thanks for your sharing. Unfortunately, I still couldn't produce the same components as you had. I doubt something wrong with my crystal symmetry. Please see my code and my resulting image below. Thanks.

![image](https://user-images.githubusercontent.com/117882416/207420460-299c9666-0a9b-4f77-8511-7a0fcdea3f02.png)

```
cs = crystalSymmetry('m-3m');
ss = specimenSymmetry('triclinic');

phi_2 = [45*degree]
opt = {'phi2',phi_2,'add2all','MarkerEdgeColor','black','maxphi1',2*pi,'MarkerSize',20};
close all;

figure;
 
ori = orientation.cube(cs,ss);
plotSection(ori, 'MarkerColor', 'darkred','Marker','s', opt{:});
hold on; 

ori = orientation.cubeND45(cs,ss)
plotSection(ori,'MarkerColor', 'steelblue','Marker','d', opt{:});
ori = orientation.byMiller([1 1 1],[1 1 2],cs);
plotSection(ori,'MarkerColor', 'gold','Marker','h',opt{:});

ori1 = orientation.byMiller([1 1 0],[1 1 1],cs);
plotSection(ori1,'MarkerColor', 'lightSkyBlue','Marker','h',opt{:});

ori2 = orientation.byMiller([1 1 0],[1 1 2],cs);
plotSection(ori2,'MarkerColor', 'r','Marker','d',opt{:});

ori = orientation.byMiller([1 1 0],[0 0 1],cs);
plotSection(ori,'MarkerColor', 'g','Marker','o',opt{:});

% 
ori = orientation.byMiller([5 5 2],[1 1 5],cs);
plotSection(ori,'MarkerColor', 'b','Marker','+',opt{:});

% ^
ori = orientation.byMiller([1 1 2],[1 1 1],cs);
plotSection(ori,'MarkerColor', 'none','Marker','^',opt{:});

% v
ori = orientation.byMiller([4 4 11],[11 11 8],cs);
plotSection(ori,'MarkerColor', 'none','Marker','v',opt{:});


hold off;  
```

**Reply by NovakEvana** [reply]
Regarding your questions, please see my comment below/

- `Your reference paper seems to have orthorhombic specimen symmetry` You can refer to the paper at: https://www.mdpi.com/1071864
![image](https://user-images.githubusercontent.com/117882416/207421986-8b586775-c371-433f-bd9a-1d85452aad74.png)

- `The definition of the alpha fibre in your reference paper is clearly wrong - its the same as the gamma fibre.` Yes, you're right. That one supposed to be zeta fiber.
- `The orange fibre in the plot is <110> || Z, however this does not coincide with the definition of the alpha fibre within MTEX.` That's zeta fiber for BCC materials. I don't know which type of material mtex referenced from.

- `The tau fibre as define in your reference paper does not coincide with the definition of the tau fibre within MTEX. Is there any reliable source on how these fibres are defined?` My majority of texture analysis refers to this book:  Engler, O., & Randle, V. (2009). Introduction to Texture Analysis: Macrotexture, Microtexture, and Orientation Mapping, Second Edition (2nd ed.). CRC Press. https://doi.org/10.1201/9781420063660,  in which I excerpted 2 images describing the texture fiber specifically for FCC and BCC:
 
![image](https://user-images.githubusercontent.com/117882416/207423439-5a405bde-4c6b-4dda-a536-bae1d9946af3.png)
![image](https://user-images.githubusercontent.com/117882416/207423554-37740348-1954-4af9-af21-2ff626cd91a4.png)

**Reply by ralfHielscher** [reply]
Hi Evana,

this is a mistake in the paper. In cubic crystal symmetry each orientation has exactly 24 crystallographic equivalent orientations.
Therefore, it is sufficient to plot not the entire Euler space with phi1 = 0..360, Phi=0..180, phi_2 = 0..360 but only the fundamental region 
phi1 = 0..360, Phi=0..90, phi_2 = 0..90. This is exactly the 1/8 of the entire space and hence each orientations appears exactly 3 times within this fundamental region. 
When plotting in Euler angles the space gets very distorted at the boundaries and hence for Phi = 0 orientations may appear more often (as you pointed out for the cube orientation). However, this is impossible for all other orientations - these should appear exactly 3 times in all sections combined. In your plot however, The yellow orientations appears 6 times alone in the 45 degree section, which indicates an additional symmetry. Also the additional 3 yellow orientations in comparison to your plot are exactly at a 180 degree rotation around ND - which is exactly a symmetry operation of orthorhombic specimen symmetry.

Thank you for the references.

Ralf.

**Reply by NovakEvana** [reply]
Hi Ralf,

You're right that there should be only 3 orientations of each component in one plot section. However, except for the Cube component, the rest of them with a number of orientations greater than 3 will be associated with a difference in phi1 of +-60 degrees (see my attached image below, the highlighted orientation is the yellow one you indicated above). I think the atomic spacing in each crystallography direction is the same so they can lump them into each other (i.e., (90 55 45) is (111)[1-21] and (30 55 45) is (111)[-1-12]; so [1-21] and [-1-12] has the same atomic spacing).

`From: https://doi.org/10.3390/ma11081337`
![image](https://user-images.githubusercontent.com/117882416/207664389-763448ca-b030-494e-937a-f5c443e874d2.png)

Thanks.

**Reply by NovakEvana** [reply]
> Hi Ralf,
> 
> Thanks for your sharing. Unfortunately, I still couldn't produce the same components as you had. I doubt something wrong with my crystal symmetry. Please see my code and my resulting image below. Thanks.
> 
> ![image](https://user-images.githubusercontent.com/117882416/207420460-299c9666-0a9b-4f77-8511-7a0fcdea3f02.png)
> 
> ```
> cs = crystalSymmetry('m-3m');
> ss = specimenSymmetry('triclinic');
> 
> phi_2 = [45*degree]
> opt = {'phi2',phi_2,'add2all','MarkerEdgeColor','black','maxphi1',2*pi,'MarkerSize',20};
> close all;
> 
> figure;
>  
> ori = orientation.cube(cs,ss);
> plotSection(ori, 'MarkerColor', 'darkred','Marker','s', opt{:});
> hold on; 
> 
> ori = orientation.cubeND45(cs,ss)
> plotSection(ori,'MarkerColor', 'steelblue','Marker','d', opt{:});
> ori = orientation.byMiller([1 1 1],[1 1 2],cs);
> plotSection(ori,'MarkerColor', 'gold','Marker','h',opt{:});
> 
> ori1 = orientation.byMiller([1 1 0],[1 1 1],cs);
> plotSection(ori1,'MarkerColor', 'lightSkyBlue','Marker','h',opt{:});
> 
> ori2 = orientation.byMiller([1 1 0],[1 1 2],cs);
> plotSection(ori2,'MarkerColor', 'r','Marker','d',opt{:});
> 
> ori = orientation.byMiller([1 1 0],[0 0 1],cs);
> plotSection(ori,'MarkerColor', 'g','Marker','o',opt{:});
> 
> % 
> ori = orientation.byMiller([5 5 2],[1 1 5],cs);
> plotSection(ori,'MarkerColor', 'b','Marker','+',opt{:});
> 
> % ^
> ori = orientation.byMiller([1 1 2],[1 1 1],cs);
> plotSection(ori,'MarkerColor', 'none','Marker','^',opt{:});
> 
> % v
> ori = orientation.byMiller([4 4 11],[11 11 8],cs);
> plotSection(ori,'MarkerColor', 'none','Marker','v',opt{:});
> 
> 
> hold off;  
> ```

Hi Ralf, 

Could you please also check on this one? 
Thanks,
Evana.

**Reply by ralfHielscher** [reply]
Hi Evana,

the point is you need to use orthorhombic specimen symmetry:
```matlab
ss = specimenSymmetry('orthorhombic');
```
Ralf.

**Reply by NovakEvana** [reply]
Thanks Ralf.
